maintainers/patches

```diff
@@ -135,6 +146,29 @@ save() {
   done
 }
 
+check() {
+  # Overwrite the commit function to do nothing.
+  commit() { true; }
+  for module in "${MODULES[@]}"; do
+    local root="$(get_root "${module}")"
+    ( set -e
+      cd third_party/"${module}"
+      git add .
+      git commit --allow-empty -qmx
+      local top="$(get_head)"
+      apply_module "${root}" "${module}"
```

Comment:
I wonder if we clutter our submodule's git that way? If I deploy a few times, is my git history of tock spammed with these commits?

Reply:
No, you get back to the exact state you were. However, the reflog will be cluttered. But this is usually only used to recover from an immediate mistake in which case the clutter is not visible.

I agree the design is not great, but that's essentially because we are not using submodules properly. We should have our own private forks.

Reply:
That's what I thought. reflog should be okay for now.

Reply:
Actually it's not exactly the same state. I don't restore the index (just the head and the working tree). Let me do it. It can be useful if someone is adding a new patch by incrementally adding to the index between deploy.